link/platform.go

```diff
@@ -0,0 +1,29 @@
+package link
+
+import "runtime"
+
+func byArch(arch_map map[string]string) string {
```

Comment:
Suggestion: just switch on runtime.GOARCH and prepend `__x64_`, etc. to a string supplied by the user.
```suggestion
func archSyscallName(name string) string {
```

```
archSyscallName("sys_getpid")
```